lib/spack/spack/modules/common.py

```diff
@@ -215,7 +215,10 @@ def root_path(name):
     """
     # Root folders where the various module files should be written
     roots = spack.config.get('config:module_roots', {})
-    path = roots.get(name, os.path.join(spack.paths.share_path, name))
+
+    module_base = os.path.join(str(spack.store.root), 'modules')
+
+    path = roots.get(name, os.path.join(module_base, name))
```

Comment:
Relating to https://github.com/spack/spack/pull/11871#discussion_r555376992, I think most of that comment could be resolved here:

* If `config.get('config:install_trees')` and `config.get('config:module_roots')` are both not empty, raise an error
* Else if `config.get('config:module_roots')` is not empty, use that
* Else if `config.get('config:install_trees')` is not empty, construct the module root from that
* (else, neither is set, in which case that should also probably be an error)

Reply:
Ah ok, thats a good idea. I'll start making those changes.

Reply:
I have made the suggested changes.